internal/http/download_test.go

```diff
@@ -143,7 +150,52 @@ func TestDownload_RedirectLoop(t *testing.T) {
 	assert.ErrorContains(t, err, "stopped after 10 redirects")
 }
 
-func startFakeDownloadServer(t *testing.T, handler http.Handler) string {
+func TestDownload_BasicAuth(t *testing.T) {
+	baseURL := startFakeDownloadServer(t, false, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if user, pass, _ := r.BasicAuth(); user != "myuser" || pass != "mypassword" {
+			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			return
+		}
+	}))
+	err := internalhttp.Download(context.TODO(), baseURL, io.Discard)
+	assert.ErrorContains(t, err, "bad status: 401 Unauthorized")
+	err = internalhttp.Download(context.TODO(), baseURL, io.Discard, internalhttp.WithBasicAuth("myuser", "mypassword"))
+	assert.NoError(t, err)
+}
+
+func TestDownload_InsecureSkipTLSVerify(t *testing.T) {
+	baseURL := startFakeDownloadServer(t, true, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("content"))
+		require.NoError(t, err)
+		w.WriteHeader(http.StatusOK)
```

Comment:
We can also just leave the handler body empty, as the content is discarded later on, and the status defaults to OK anyways.

(The call to WriteHeader has no effect anyways, as it's happening after the first write.)

```suggestion
```

Reply:
That makes sense. I have incorporated the proposed changes on my branch.